Assets/uAdventureAnalytics/Plugins/Simva/SimvaController.cs

```diff
@@ -0,0 +1,470 @@
+            yield return new Tuple<string, string>("{\"message\": \"Unable to connect\"}", null);
+        }
+        else
+        {
+            if (www.responseCode != 200)
+            {
+                yield return new Tuple<string, string>(www.downloadHandler.text, null);
+            }
+            else
+            {
+                yield return new Tuple<string, string>(null, www.downloadHandler.text);
+            }
+        }
+    }
+
+    public IEnumerator setResults(string activityId, string results, bool tofile)
+    {
+        JSONNode body = new JSONClass();
+        body.Add("result", results);
+        body.Add("tofile", new JSONData(tofile));
+        string bodystring = body.ToJSON(0);
+
+        UnityWebRequest www = new UnityWebRequest(this.URL + "/activities/" + activityId + "/result", "POST");
+        byte[] bodyRaw = System.Text.Encoding.UTF8.GetBytes(bodystring);
+        www.uploadHandler = (UploadHandler)new UploadHandlerRaw(bodyRaw);
+        www.downloadHandler = (DownloadHandler)new DownloadHandlerBuffer();
+
+        www.SetRequestHeader("Content-Type", "application/json");
+        www.SetRequestHeader("Authorization", "Bearer " + this.jwt);
+
+        yield return www.SendWebRequest();
+
+        if (www.isNetworkError)
+        {
+            yield return new Tuple<string, string>("{\"message\": \"Unable to connect\"}", null);
+        }
+        else
+        {
+            if (www.responseCode != 200)
+            {
+                yield return new Tuple<string, string>(www.downloadHandler.text, null);
+            }
+            else
+            {
+                yield return new Tuple<string, string>(null, www.downloadHandler.text);
+            }
+        }
+    }
+
+    public IEnumerator setCompletion(string activityId, bool completion)
+    {
+        JSONNode body = new JSONClass();
+        body.Add("status", new JSONData(completion));
+        string bodystring = body.ToString();
+
+        UnityWebRequest www = new UnityWebRequest(this.URL + "/activities/" + activityId + "/completion", "POST");
+        byte[] bodyRaw = System.Text.Encoding.UTF8.GetBytes(bodystring);
+        www.uploadHandler = (UploadHandler)new UploadHandlerRaw(bodyRaw);
+        www.downloadHandler = (DownloadHandler)new DownloadHandlerBuffer();
+
+        www.SetRequestHeader("Content-Type", "application/json");
+        www.SetRequestHeader("Authorization", "Bearer " + this.jwt);
+
+        yield return www.SendWebRequest();
+
+        if (www.isNetworkError)
+        {
+            yield return new Tuple<string, string>("{\"message\": \"Unable to connect\"}", null);
+        }
+        else
+        {
+            if (www.responseCode != 200)
+            {
+                yield return new Tuple<string, string>(www.downloadHandler.text, null);
+            }
+            else
+            {
+                yield return new Tuple<string, string>(null, www.downloadHandler.text);
+            }
+        }
+    }
```

Comment:
All these code seems to be duplicated following the same structure and request handling. I suggest using another coroutine for response handling

Reply:
Fixed with ProcessWWW method